### ISSUE TITLE
Visiting a webapp with syntax error

### ISSUE THREAD

**Issue by Timer** [vercel/next.js#14728]
# Bug report

## Describe the bug

Dual syntax error when visiting app from server with a compilation error:

```
error - ./pages/index.js:56:4
Syntax error: Unexpected token, expected "jsxTagEnd"
  54 |         </button>
  55 |       </div
> 56 |     </main>
     |     ^
  57 |   )
  58 | }
  59 | 
Error: Syntax error: Unexpected token, expected "jsxTagEnd"
  54 |         </button>
  55 |       </div
> 56 |     </main>
     |     ^
  57 |   )
  58 | }
  59 | 
    at getBabelError (/Users/joe/Documents/Development/Work/zeit/next.js/packages/next/dist/build/webpack/plugins/wellknown-errors-plugin/parseBabel.js:4:91)
    at getModuleBuildError (/Users/joe/Documents/Development/Work/zeit/next.js/packages/next/dist/build/webpack/plugins/wellknown-errors-plugin/webpackModuleError.js:1:2469)
    at /Users/joe/Documents/Development/Work/zeit/next.js/packages/next/dist/build/webpack/plugins/wellknown-errors-plugin/index.js:1:549
    at Array.map (<anonymous>)
    at /Users/joe/Documents/Development/Work/zeit/next.js/packages/next/dist/build/webpack/plugins/wellknown-errors-plugin/index.js:1:478
    at SyncHook.eval [as call] (eval at create (/Users/joe/Documents/Development/Work/zeit/next.js/node_modules/tapable/lib/HookCodeFactory.js:19:10), <anonymous>:7:1)
    at SyncHook.lazyCompileHook (/Users/joe/Documents/Development/Work/zeit/next.js/node_modules/tapable/lib/Hook.js:154:20)
    at Compilation.seal (/Users/joe/Documents/Development/Work/zeit/next.js/node_modules/webpack/lib/Compilation.js:1284:19)
    at /Users/joe/Documents/Development/Work/zeit/next.js/node_modules/webpack/lib/Compiler.js:675:18
    at /Users/joe/Documents/Development/Work/zeit/next.js/node_modules/webpack/lib/Compilation.js:1261:4
event - compiled successfully
```

## To Reproduce

1. Create a syntax error.
1. Start dev server.
1. Visit page URL.
1. Observe error in the terminal

## Expected behavior

Only once:

```
error - ./pages/index.js:56:4
Syntax error: Unexpected token, expected "jsxTagEnd"
  54 |         </button>
  55 |       </div
> 56 |     </main>
     |     ^
  57 |   )
  58 | }
  59 | 
```

**Comment by jevangelista0** [comment]
Hello, it's my first time participating in Hacktoberfest and I'm still figuring out how contributing works.

I tried triggering the bug and it looks like it only shows up once. Am I doing something wrong or should this issue already be resolved?

**Comment by mayank-budhiraja** [comment]
@jvincent-dev this issue seems to be resolved already

**Comment by huozhi** [comment]
NOTE: If you're using default swc compiler this won't be an issue any more, only preserved when using babel.

**Comment by timneutkens** [comment]
Going to close this as it is taken care of when using SWC 👍

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
